comms/dht/src/storage/error.rs

```diff
@@ -42,4 +43,12 @@ pub enum StorageError {
     MessageFormatError(#[from] MessageFormatError),
     #[error("Unexpected result: {0}")]
     UnexpectedResult(String),
+    #[error("Diesel R2d2 error: `{0}`")]
+    DieselR2d2Error(#[from] SqliteStorageError),
+}
+
+impl PartialEq for StorageError {
```

Comment:
I think you can use the `PartialEq` derive macro

Reply:
The `PartialEq` derive macro cannot be applied to some of the members here, i.e. `tokio::task::JoinError` and `tari_utilities::message_format::MessageFormatError`.

Reply:
I suspect this results in an infinite recursion - by doing `self == other` you can calling `PartialEq::eq`. 
Try adding this to the bottom and it should stack overflow
```rust

#[cfg(test)]
mod test {
    use super::*;

    #[test]
    fn will_break() {
        let _ = StorageError::InvalidUnicodePath == StorageError::InvalidUnicodePath;
    }
}
```

Reply:
If you want to check equality of an error in tests without implementing `PartialEq` you can use the `matches!`, a `match` or `tari_test_utils::unpack_enum!`  or remove if unused.

Reply:
Was actually not needed here, but I had to change it for `impl PartialEq for WalletStorageError`, also added a comparison test for `WalletStorageError`.